docs/source/webui.rst

```diff
@@ -0,0 +1,118 @@
+
+It can also be installed by extracting the latest tar-ball from https://ops.stackstorm.net/releases/st2/0.8dev/webui/ into `/opt/stackstorm/static/webui`.
+
+This type of installation has no additional requirements.
+
+Development
+-----------
+
+With production version it is almost impossible to make any changes in code, besides configuration. In case any changes are needed, development version has to be installed.
+
+First of all, you need to make sure you have latest stable node and npm packages installed.
+
+::
+
+   $ node -v
+   v0.10.32
+
+   $ npm -v
+   1.4.9
+
+then you need to globally install bower and gulp.
+
+::
+
+   $ npm install -g bower
+   $ npm install -g gulp
+
+
+Clone the latest version of st2web repository
+
+::
+
+   $ git clone https://github.com/StackStorm/st2web.git
+   $ cd st2web
+
+then you need to install the requirements
+
+::
+
+   $ npm install
+   $ bower install
+
+and finally run build system to lint js files, compile css and watch for changes
+
+::
+
+   $ gulp
+
+At that point you should be able to point your browser to http://localhost:3000/ and see the the page. Every change in code would be automatically recompiled.
+
+Gulp has several additional options:
+
+* `gulp build` - just lint and compile everything
+* `gulp test` - build the project and then run e2e tests
+* `gulp serve` - build the project and start serving it at 3000 port
+* `gulp production` - build production version of the project
+
+The production version of the code is located in the `build/` folder. It consists of compiled and minified versions of JS and CSS files, doesn't have any additional dependencies and can be deployed to any server that can serve static files. The version produced by `gulp production` is an exact copy of the data used in Express installation and can be deployed the same way.
+
+Configuration
+-------------
+
+For UI to work properly, both client and server side should be configured properly.
+
+On a UI side, there is a file `config.js` in a root of the project which contains the list of servers this UI can connect to. The file consists of an array of objects, each have a `name`, `url` and `auth` properties.
+
+::
+
+   hosts: [{
+     name: 'Express Deployment',
+     url: 'http://172.168.90.50:9101',
+     auth: true
+   },{
+     name: 'Development Environment',
+     url: 'http://172.168.50.50:9101'
+   }]
+
+
+Multiple servers could be configured for user to pick from. To disconnect from the current server and return to login screen, pick 'Disconnect' from the drop down at the top right corner of the UI.
+
+On an ST2 side, `CORS <https://en.wikipedia.org/wiki/Cross-origin_resource_sharing>`__ should also be properly configured. In st2.conf, `allow_origin` property of the [api] section should contain the Origin header browser sends with every request. For example, if you have deployed UI on its own server and accessing it using url `http://webui.example.com:8000`, your config should look like that:
```

Comment:
Might be good to add that the origin URL must **not** contain a trailing slash (I personally spent a couple of minutes debugging / dealing with that).